server/services/utils/helpers.js

```diff
@@ -26,3 +26,38 @@ export const isIndexNotFoundError = (err) => {
       'Configured indices are not found: [.opendistro-alerting-config]'
   );
 };
+
+export function getClientBasedOnDataSource(
```

Comment:
Can you instead create a base class and have other services extend that. It removes a lot of redundant code that way.

Reply:
addressed